haskell/test/SMT/Model/BaseDAO/Management.hs

```diff
@@ -0,0 +1,76 @@
+-- SPDX-FileCopyrightText: 2021 TQ Tezos
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+module SMT.Model.BaseDAO.Management
+  ( applyAcceptOwnership
+  , applyTransferOwnership
+  , applyCallCustom
+  ) where
+
+import Universum
+
+import Control.Monad.Except (throwError)
+import qualified Data.Map as M
+
+import Michelson.Text (unsafeMkMText)
+import qualified Michelson.Typed as T
+import Lorentz hiding (not, cast, checkSignature, get)
+
+import Ligo.BaseDAO.Types
+import Test.Ligo.RegistryDAO.Types
+import SMT.Model.BaseDAO.Types
+
+applyTransferOwnership :: ModelCall -> TransferOwnershipParam -> ModelT ()
+applyTransferOwnership mc param = do
+  selfAddr <- get <&> msSelfAddress
+
+  store <- getStore
+  unless (mcSender mc == sAdmin store) $ throwError NOT_ADMIN
+
+  let newOwner = arg #newOwner param
+
+  modifyStore \s ->
+    if selfAddr == newOwner then s { sAdmin = newOwner }
+    else s { sPendingOwner = newOwner}
+
+applyAcceptOwnership :: ModelCall -> ModelT ()
+applyAcceptOwnership mc = do
+  store <- getStore
+  if (store & sPendingOwner) == (mc & mcSender) then
+    modifyStore $ \s -> s { sAdmin = (mc & mcSender)}
+  else
+    throwError NOT_PENDING_ADMIN
+
+applyCallCustom :: ContractType -> ModelCall -> CallCustomParam -> ModelT ()
+applyCallCustom ct _ (epName, packedParam) = do
+  case ct of
+    BaseDaoContract -> throwError ENTRYPOINT_NOT_FOUND
+
+    RegistryDaoContract ->
+      if (epName == unsafeMkMText "receive_xtz") then
+        pure ()
+      else if (epName == unsafeMkMText "lookup_registry") then do
+        store <- getStore
+
+        let (unpackVal :: Either ModelError (MText, Address, Map MText MText)) = do
+              LookupRegistryParam key addr <- first (const UNPACKING_FAILED) $ lUnpackValueRaw @LookupRegistryParam packedParam
+              packedRegistry <- case M.lookup key (store & sExtra & unDynamic & T.unBigMap ) of
+                Just val -> Right val
+                Nothing -> Left MISSING_VALUE
+              registry <- first (const UNPACKING_FAILED) $ lUnpackValueRaw @(Map MText MText) packedRegistry
+              Right (key, addr, registry)
+        case unpackVal of
+          Right (key, addr, registry) -> do
+            let val = M.lookup key registry
+            -- Note: Currently, we don't check the storage of the view contract
```

Comment:
Why not?

Reply:
@pasqu4le I've added storage check for a view contract now. This contract storage is mainly used by `lookup_registry` entrypoints. This should be ready for re-review.